code/modules/events/vent_clog.dm

```diff
@@ -26,10 +28,11 @@
 
 /datum/round_event/vent_clog/tick()
 	if(activeFor % interval == 0)
-		var/obj/vent = pick_n_take(vents)
+		var/obj/machinery/atmospherics/components/unary/vent = pick_n_take(vents)

```

Comment:
You're removing a vent from the vents list but then the line just below you change vent to be another vent without having used the first vent.